modules/user_accounts/php/edit_user.class.inc

```diff
@@ -909,40 +909,68 @@ class Edit_User extends \NDB_Form
         // get the editor's permissions
         $perms = $editor->getPermissionsVerbose();
 
-        $lastRole = '';
-        $group    = [];
-        foreach ($perms as $row) {
-            if ($row['type'] != $lastRole) {
-                $lastRole = $row['type'];
-                $group[]  = $this->form->createElement(
-                    'static',
-                    '',
-                    '',
-                    '</div>'
-                    . "<h3 id=\"header_$lastRole\" "
-                    . "class=\"perm_header button\" "
-                    . "style=\"text-align: center; margin-top: 5px;\">"
-                    .ucwords($row['type'])
-                    . '</h3>'
-                    . "<div id=\"perms_$lastRole\" style=\"margin-top: 5px;\">"
-                );
+        usort(
+            $perms,
+            function ($a, $b) {
+                return strcmp($a['label'], $b['label']);
             }
-
-            // If the user is editing his/her own account, disable all permission
-            // checkboxes. Note that they will not be submitted when the form is
-            // saved
-            $attribs = ["class" => "perm_$lastRole"];
+        );
+        $permissions = [];
+        $roles       = [];
+        foreach ($perms as $row) {
+            $lastRole = $row['type'];
+            $attribs  = ["class" => "perm_$lastRole"];
             if ($this->_isEditingOwnAccount()) {
                 $attribs['disabled'] = true;
             }
-            $group[] = $this->createCheckbox(
-                'permID['.$row['permID'].']',
-                htmlspecialchars($row['label']) . "<br>",
-                $attribs
-            );
+            if ($row['type'] === 'Roles') {
+                $roles[] = $this->createCheckbox(
+                    'permID['.$row['permID'].']',
+                    htmlspecialchars($row['label']) . "<br>",
+                    $attribs
+                );
+            } else {
+                $permissions[] = $this->createCheckbox(
+                    'permID['.$row['permID'].']',
+                    htmlspecialchars($row['label']) . "<br>",
+                    $attribs
+                );
+            }
         }
-        $this->addGroup($group, 'PermID_Group', 'Permissions', "");
-        unset($group);
+
+        $collection   = [];
+        $collection[] = $this->form->createElement(
+            'static',
+            '',
+            '',
+            '</div>'
+                        . "<h3 id=\"header_Permission\" "
```

Comment:
These fields you are hardcoding here are actually dynamic values read from the permission_category table in the dtabase

See RB example https://github.com/aces/Loris/blob/main/raisinbread/RB_files/RB_permissions_category.sql

so that's not a valid change